### PR TITLE
[SYCL] Add LIT regression test for ambiguous call to `memcpy`

### PR DESCRIPTION
**Pull request by uditagarwal97**
`detail::memcpy`, even though in a different namespace, can cause ambiguity with libc's memcpy, due to argument dependent lookup (ADL). This compile-only
test checks for ambiguity in call to `detail::memcpy`.

As discussed in https://github.com/intel/llvm/pull/14836#issuecomment-2266136834

### CODE REVIEW

File: sycl/test/regression/ambigious-memcpy.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// detail::memcpy, even though in a different namespace, can cause ambiguity with libc's memcpy, due to argument dependent lookup (ADL). This compile-only test checks for ambigiouity in call to detail::memcpy.
```
Comment (AlexeySachkov) [inline sycl/test/regression/ambigious-memcpy.cpp:1]: Should not be longer than 80 symbols

File: sycl/test/regression/ambigious-memcpy.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+// RUN: %clang -fsycl -fsycl-device-only %s -o %s.out
+
+#include <sycl/vector.hpp>
+
+template <typename T>
+void foo(T *dst, T *src, size_t count) {
+	          memcpy(dst, src, count * sizeof(T));
```
Comment (AlexeySachkov) [inline sycl/test/regression/ambigious-memcpy.cpp:9]: 2-4 spaces is more than enough :)

File: sycl/test/regression/ambigious-memcpy.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// detail::memcpy, even though in a different namespace, can cause ambiguity with libc's memcpy, due to argument dependent lookup (ADL). This compile-only test checks for ambigiouity in call to detail::memcpy.
+
+// RUN: %clang -fsycl -fsycl-device-only %s -o %s.out
```
Comment (AlexeySachkov) [inline sycl/test/regression/ambigious-memcpy.cpp:3]: I think that you can limit this to `-fsyntax-only` and therefore drop `-o`

**Review by AlexeySachkov — COMMENTED** [review]
Thanks for addressing the feedback!

File: sycl/test/regression/ambigious-memcpy.cpp (hunk 3 above)
Comment (uditagarwal97) [reply sycl/test/regression/ambigious-memcpy.cpp]: Fixed in 87f8f3ea47d9c9ec5ea65471fca9bb168f2b262f

File: sycl/test/regression/ambigious-memcpy.cpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// detail::memcpy, even though in a different namespace, can cause ambiguity
+// with libc's memcpy, due to argument dependent lookup (ADL). This compile-only
+// test checks for ambigiouity in call to detail::memcpy.
+
+// RUN: %clang -fsycl -fsycl-device-only -fsyntax-only %s
```
Comment (steffenlarsen) [inline sycl/test/regression/ambigious-memcpy.cpp:5]: Tiny nit and by no means blocking, but normally we have the REQUIRES and RUN lines first and the description after.

```suggestion
// RUN: %clang -fsycl -fsycl-device-only -fsyntax-only %s

// detail::memcpy, even though in a different namespace, can cause ambiguity
// with libc's memcpy, due to argument dependent lookup (ADL). This compile-only
// test checks for ambiguity in call to detail::memcpy.
```

Note also a small spelling mistake fix in there: "ambigiouity" -> "ambiguity".

**Review by steffenlarsen — APPROVED** [review]
LGTM! For the sake of history, would you mind adding a summary of the motivation? It does not have to be as detailed as in https://github.com/intel/llvm/pull/14836 and could just be a copy of the comment from the test.

**Review by AlexeySachkov — APPROVED** [review]

**Comment by uditagarwal97** [comment]
@intel/llvm-gatekeepers the PR is ready to be merged.

**Pull request merged**